test/parsing/ferguson/this-intent.chpl

```diff
@@ -0,0 +1,22 @@
+record R {
+  var x:int;
+  proc f() {
+    writeln(x);
+  }
+  proc ref g() {
+    writeln(x);
+  }
+  proc const h() {
+    writeln(x);
+  }
+  proc const ref i() {
+    writeln(x);
+  }
```

Comment:
No, because we don't check const-ness for such things right yet. This PR started as taking some separable things out of PR #5134. But the parent PR should enable the const-ness checking. I'll add a test there.